app/models/pin.rb

```diff
@@ -1,4 +1,5 @@
 class Pin < ActiveRecord::Base
      belongs_to :user
      has_attached_file :image, :styles => { :medium => "300x300>", :thumb => "100x100>" }
+     validates_attachment_content_type :image, :content_type => ["image/jpg", "image/jpeg", "image/png", "image/gif"]
```

Comment:
This is the validation rule which checks that the file being uploaded has one of the standard image content-types, so that someone would be prevented from uploaded a pdf for example. You'd probably want to add additional validation rules, like on the max file size, for example.